docs/how-to/hip_porting_driver_api.rst

```diff
@@ -0,0 +1,361 @@
+HIP provides a context API to facilitate easy porting from existing Driver codes.
+In HIP, the ``Ctx`` functions largely provide an alternate syntax for changing the active device.
+
+Most new applications will prefer to use ``hipSetDevice`` or the stream APIs, therefore HIP has marked ``hipCtx`` APIs as **deprecated**. Support for these APIs may not be available in future releases. For more details on deprecated APIs please refer :doc:`reference/deprecated_api_list`.
+
+HIP Module and ``Ctx`` APIs
+===========================
+
+Rather than present two separate APIs, HIP extends the HIP API with new APIs for Modules and ``Ctx`` control.
+
+``hipModule`` API
+-----------------
+
+Like the CUDA Driver API, the Module API provides additional control over how code is loaded, including options to load code from files or from in-memory pointers.
+NVCC and HIP-Clang target different architectures and use different code object formats: NVCC is ``cubin`` or ``ptx`` files, while the HIP-Clang path is the ``hsaco`` format.
+The external compilers which generate these code objects are responsible for generating and loading the correct code object for each platform.
+Notably, there is not a fat binary format that can contain code for both NVCC and HIP-Clang platforms. The following table summarizes the formats used on each platform:
+
+.. list-table:: Module formats
+   :header-rows: 1
+
+   * - Format
+     - APIs
+     - NVCC
+     - HIP-CLANG
+   * - Code Object
+     - ``hipModuleLoad``, ``hipModuleLoadData``
+     - ``.cubin`` or PTX text
+     - ``.hsaco``
+   * - Fat Binary
+     - ``hipModuleLoadFatBin``
+     - ``.fatbin``
+     - ``.hip_fatbin``
+
+``hipcc`` uses HIP-Clang or NVCC to compile host codes. Both of these may embed code objects into the final executable, and these code objects will be automatically loaded when the application starts.
+The ``hipModule`` API can be used to load additional code objects, and in this way provides an extended capability to the automatically loaded code objects.
+HIP-Clang allows both of these capabilities to be used together, if desired. Of course it is possible to create a program with no kernels and thus no automatic loading.
+
+``hipCtx`` API
+--------------
+
+HIP provides a ``Ctx`` API as a thin layer over the existing Device functions. This ``Ctx`` API can be used to set the current context, or to query properties of the device associated with the context.
+The current context is implicitly used by other APIs such as ``hipStreamCreate``.
+
+hipify translation of CUDA Driver API
+-------------------------------------
+
+The HIPIFY tools convert CUDA Driver APIs for streams, events, modules, devices, memory management, context, profiler to the equivalent HIP driver calls. For example, ``cuEventCreate`` will be translated to ``hipEventCreate``.
```

Comment:
Aren't the hip calls just the (extended) HIP runtime API calls? I'm not aware of a HIP driver API

Reply:
Yes, they are. 

This section refers to HIPIFY. It's a tool that converts a CUDA source code to HIP, a kind of transpiler.

Reply:
Then this can't translate to the equivalent HIP "driver" calls, but just to the equivalent HIP calls

Reply:
Removed the driver world.